### PR TITLE
Add Updater

### LINKED ISSUE #534: Engine - Make UI for Auto Updater
|    DOD    | Done |    Asignees    | Branch | PR |
|-----------|------|----------------|-------|-----|
| Status    | ❌ | <assignee> | <branchName> | <number> |

### PR DESCRIPTION
**Pull request by nicolasespinoza**
### Add Updater
- Added and slightly refactored Updater from previous Unity Synthesis ([1](https://github.com/Autodesk/synthesis/blob/master/engine/Assets/Scripts/Engine/Utilities/VersionManager.cs)) ([2](https://github.com/Autodesk/synthesis/blob/master/engine/Assets/Scripts/Engine/Utilities/UnityHandler.cs))
- Added Dialog UI for notification which opens respective update version URL

Annotation 2020-09-09 164547

**How to test**
One way to test the functionality updater is to invert the if statement [here](https://github.com/Autodesk/synthesis/blob/54c8d70f05bbd3e64395736b4897c1f262e90a2d/modules/SynthesisCore/UI/Ui.cs#L85), which will display the dialog. Please note the information there will be invalid though because of the inverted if statement. This will not happen in production.

Closes #534

### CODE REVIEW

File: modules/SynthesisCore/Utilities/Updater.cs [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+                using (client.OpenRead("https://raw.githubusercontent.com/Autodesk/synthesis/master/VersionManager.json"))
+                {
+                    //Logger.Log("[Updater] Connection to version checker established");
+                    return true;
+                }
+            }
+            catch
+            {
+                Logger.Log("[Updater] Connection to version checker failed", LogLevel.Warning);
```
Comment (matthew17754) [inline modules/SynthesisCore/Utilities/Updater.cs:57]: `Connection to update servers failed` would be more fitting

**Review by matthew17754 — CHANGES_REQUESTED** [review]
We should also add a reference to display the installed version number somewhere else in the GUI, such as the bottom left of the Settings or Help menus. Also some small suggestions below.

**Review by KiraCorbett — APPROVED** [review]
Looks good. Thanks for adding the versioning number too.

**Review by matthew17754 — APPROVED** [review]
Looks good. 👍

**Pull request merged**